### DISCUSSION TITLE
Dependency Injection with getLoadContext for default Remix server

### DISCUSSION THREAD

**Discussion by tatemz** [remix-run/remix#5263] [Proposals]
`remix dev` is great. `getLoadContext` is great. I wish I could add context properties without having to run my own express server, which removes from the benefits of the default express server that comes with `remix dev` :( ([relevant lines](https://github.com/remix-run/remix/blob/main/packages/remix-serve/index.ts#L26-L35))

I'd love for a way to plug into `getLoadContext` in `entry.server.ts`

**Comment by sergiodxa** [comment]
The getLoadContext can't live in entry.server as entry.server is part of the Remix app but getLoadContext is an HTTP server adapter feature to let the HTTP server inject values into the Remix app, so the only way to leverage it is by using a custom HTTP server.

**Reply by belgamo** [reply]
What about middlewares? Would you recommend that to inject api gateway instances into loaders? I got it working using remix-express-vite-plugin as demonstrated [here](https://github.com/kiliman/remix-express-vite-plugin/blob/0cd1e0e2c5cd003c5e23eaa08c577069bf4ac259/example/remix-vite-express/app/root.tsx#L30)

**Reply by sergiodxa** [reply]
There's no middlewares in Remix, once that's supported it may be a way.

**Reply by belgamo** [reply]
> There's no middlewares in Remix, once that's supported it may be a way.

Not officially, right? Cause [remix-create-express-app](https://www.npmjs.com/package/remix-create-express-app) enables us to do that.

**Reply by akomm** [reply]
> The getLoadContext can't live in entry.server as entry.server is part of the Remix app but getLoadContext is an HTTP server adapter feature to let the HTTP server inject values into the Remix app, so the only way to leverage it is by using a custom HTTP server.

it does not stop the adapter from using the exported `handleRequest`. 
What does stop it from allowing an export:
`export function getContext(...)`

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
